packages/api/__tests__/integration.test.js

```diff
@@ -0,0 +1,150 @@
+const api = require('../src');
+
+const fileUploads = require('@readme/oas-examples/3.0/json/file-uploads.json');
+
+describe('`application/x-www-form-urlencoded`', () => {
+  const usptoSpec = JSON.parse(JSON.stringify(require('@readme/oas-examples/3.0/json/uspto.json')));
+  usptoSpec.servers[0].url = '{scheme}://httpbin.org/anything';
+
+  it('should support `application/x-www-form-urlencoded` requests', async () => {
+    const body = {
+      criteria: 'propertyName:value',
+    };
+
+    const metadata = {
+      dataset: 'v1',
+      version: 'oa_citations',
+    };
+
+    await expect(api(usptoSpec).post('/{dataset}/{version}/records', body, metadata)).resolves.toStrictEqual({
+      args: {},
+      data: '',
+      files: {},
+      form: {
+        criteria: 'propertyName:value',
+      },
+      headers: expect.objectContaining({
+        'Content-Type': 'application/x-www-form-urlencoded',
+        'User-Agent': expect.stringMatching(/api \(node\)\/\d+.\d+.\d+/),
+      }),
+      json: null,
+      method: 'POST',
+      origin: expect.any(String),
+      url: 'https://httpbin.org/anything/v1/oa_citations/records',
+    });
+  });
+
+  // it.skip('should send along required parameters if not supplied', async () => {
```

Comment:
We don't support sending required props that have defaults right now. Ticketed to RM-3369.

Reply:
This test also uncovered a problem with `prepareParams` where the metadata that we're sending here for the path is also being sent in the form payload:

```json
{
  args: {},
  data: '',
  files: {},
  form: { dataset: 'v1', version: 'oa_citations' },
  headers: {
    Accept: 'application/json',
    'Accept-Encoding': 'gzip,deflate',
    'Content-Length': '31',
    'Content-Type': 'application/x-www-form-urlencoded',
    Host: 'httpbin.org',
    'User-Agent': 'api (node)/4.2.0',
    'X-Amzn-Trace-Id': 'REDACTED'
  },
  json: null,
  method: 'POST',
  origin: 'REDACTED',
  url: 'https://httpbin.org/anything/v1/oa_citations/records'
}
```